controllers/k6_start.go

```diff
@@ -31,65 +28,65 @@ func isServiceReady(log logr.Logger, service *v1.Service) bool {
 func StartJobs(ctx context.Context, log logr.Logger, k6 *v1alpha1.K6, r *K6Reconciler) (ctrl.Result, error) {
 	log.Info("Waiting for pods to get ready")
 
-	err := wait.PollImmediate(time.Second*5, time.Second*60, func() (done bool, err error) {
-		selector := labels.SelectorFromSet(map[string]string{
-			"app":    "k6",
-			"k6_cr":  k6.Name,
-			"runner": "true",
-		})
-
-		opts := &client.ListOptions{LabelSelector: selector, Namespace: k6.Namespace}
-		pl := &v1.PodList{}
-		if e := r.List(ctx, pl, opts); e != nil {
-			log.Error(e, "Could not list pods")
-			return false, e
-		}
-
-		var count int
-		for _, pod := range pl.Items {
-			if pod.Status.Phase != "Running" {
-				continue
-			}
-			count++
-		}
+	selector := labels.SelectorFromSet(map[string]string{
+		"app":    "k6",
+		"k6_cr":  k6.Name,
+		"runner": "true",
+	})
 
-		log.Info(fmt.Sprintf("%d/%d pods ready", count, k6.Spec.Parallelism))
+	opts := &client.ListOptions{LabelSelector: selector, Namespace: k6.Namespace}
+	pl := &v1.PodList{}
+	if e := r.List(ctx, pl, opts); e != nil {
+		log.Error(e, "Could not list pods")
+		return ctrl.Result{}, e
+	}
 
-		if count != int(k6.Spec.Parallelism) {
-			return false, nil
+	var count int
+	for _, pod := range pl.Items {
+		if pod.Status.Phase != "Running" {
+			continue
 		}
+		count++
+	}
+
+	log.Info(fmt.Sprintf("%d/%d runner pods ready", count, k6.Spec.Parallelism))
 
-		var hostnames []string
+	if count != int(k6.Spec.Parallelism) {
+		return ctrl.Result{}, nil
+	}
 
-		sl := &v1.ServiceList{}
+	var hostnames []string
+	var err error
+	sl := &v1.ServiceList{}
 
-		if e := r.List(ctx, sl, opts); e != nil {
-			log.Error(e, "Could not list services")
-			return false, e
-		}
+	if e := r.List(ctx, sl, opts); e != nil {
+		log.Error(e, "Could not list services")
+		return ctrl.Result{}, e
+	}
 
-		for _, service := range sl.Items {
-			hostnames = append(hostnames, service.Spec.ClusterIP)
+	for _, service := range sl.Items {
+		hostnames = append(hostnames, service.Spec.ClusterIP)
 
-			if !isServiceReady(log, &service) {
-				log.Info(fmt.Sprintf("%v service is not ready, aborting", service.ObjectMeta.Name))
-				return false, nil
-			}
+		if !isServiceReady(log, &service) {
+			log.Info(fmt.Sprintf("%v service is not ready, aborting", service.ObjectMeta.Name))
+			return ctrl.Result{}, nil
```

Comment:
In case of failing service, let's requeue the event:
```suggestion
			return ctrl.Result{RequeueAfter: time.Second}, nil
```